### ISSUE TITLE
[New] Parse raw, byte and formatted strings

### ISSUE THREAD

**Issue by Thirumalai-Shaktivel** [lcompilers/lpython#512]
Example 1:
```python
print(f"{__file__} executed in {elapsed:0.2f} seconds.")
```
Error:
```console
$ lpython --show-ast --new-parser examples/expr2.py
syntax error: Token '"{__file__} executed in {elapsed:0.2f} seconds."' (of type 'string') is unexpected here
 --> examples/expr2.py:1:8
  |
1 | print(f"{__file__} executed in {elapsed:0.2f} seconds.")
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 


Note: if any of the above error or warning messages are not clear or are lacking
context please report it to us (we consider that a bug that needs to be fixed).
```
Example 2:
```python
b"I made it!"
```
Error:
```console
$ lpython --show-ast --new-parser examples/expr2.py
syntax error: Token '"I made it!"' (of type 'string') is unexpected here
 --> examples/expr2.py:1:2
  |
1 | b"I made it!"
  |  ^^^^^^^^^^^^ 


Note: if any of the above error or warning messages are not clear or are lacking
context please report it to us (we consider that a bug that needs to be fixed).
```
Example 3:
```python
r"""
Text
"""
```
Error:
```console
$ lpython --show-ast --new-parser examples/expr2.py
syntax error: Token '"""
Text
"""' (of type 'string') is unexpected here
 --> examples/expr2.py:1:2 - 3:3
  |
1 |    r"""
  |     ^^^...
...
  |
3 |    """
  | ...^^^ 


Note: if any of the above error or warning messages are not clear or are lacking
context please report it to us (we consider that a bug that needs to be fixed).
```

**Comment by Thirumalai-Shaktivel** [comment]
Another Issue in Old parser:
Using Example 1:
```console
$ lpython --show-ast examples/expr2.py
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  Binary file "/home/thirumalai/Open_Source/lpython/src/bin/lpython", in _start()
  File "./csu/../csu/libc-start.c", line 392, in __libc_start_main_impl()
  File "./csu/../sysdeps/nptl/libc_start_call_main.h", line 58, in __libc_start_call_main()
  File "/home/thirumalai/Open_Source/lpython/src/bin/lpython.cpp", line 781, in ??
    return emit_ast(arg_file, runtime_library_dir, compiler_options);
  File "/home/thirumalai/Open_Source/lpython/src/bin/lpython.cpp", line 122, in ??
    al, runtime_library_dir, infile, diagnostics, compiler_options.new_parser);
  File "/home/thirumalai/Open_Source/lpython/src/lpython/parser/parser.cpp", line 116, in LFortran::parse_python_file(Allocator&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, LFortran::diag::Diagnostics&, bool)
    ast = LPython::deserialize_ast(al, input);
  File "/home/thirumalai/Open_Source/lpython/src/lpython/python_serialization.cpp", line 44, in LFortran::LPython::deserialize_ast(Allocator&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    return v.deserialize_node();
  File "/home/thirumalai/Open_Source/lpython/src/lpython/python_ast.h", line 9730, in LFortran::LPython::AST::DeserializationBaseVisitor<LFortran::LPython::ASTDeserializationVisitor>::deserialize_node()
    case (LPython::AST::astType::mod) : return self().deserialize_mod();
  File "/home/thirumalai/Open_Source/lpython/src/lpython/python_ast.h", line 9800, in LFortran::LPython::AST::DeserializationBaseVisitor<LFortran::LPython::ASTDeserializationVisitor>::deserialize_mod()
    case (LPython::AST::modType::Module) : return self().deserialize_Module();
  File "/home/thirumalai/Open_Source/lpython/src/lpython/python_ast.h", line 9750, in LFortran::LPython::AST::DeserializationBaseVisitor<LFortran::LPython::ASTDeserializationVisitor>::deserialize_Module()
    v_body.push_back(al, LPython::AST::down_cast<LPython::AST::stmt_t>(self().deserialize_stmt()));
  File "/home/thirumalai/Open_Source/lpython/src/lpython/python_ast.h", line 10327, in LFortran::LPython::AST::DeserializationBaseVisitor<LFortran::LPython::ASTDeserializationVisitor>::deserialize_stmt()
    case (LPython::AST::stmtType::Assign) : return self().deserialize_Assign();
  File "/home/thirumalai/Open_Source/lpython/src/lpython/python_ast.h", line 9953, in LFortran::LPython::AST::DeserializationBaseVisitor<LFortran::LPython::ASTDeserializationVisitor>::deserialize_Assign()
    m_value = LPython::AST::down_cast<LPython::AST::expr_t>(self().deserialize_expr());
  File "/home/thirumalai/Open_Source/lpython/src/lpython/python_ast.h", line 10792, in LFortran::LPython::AST::DeserializationBaseVisitor<LFortran::LPython::ASTDeserializationVisitor>::deserialize_expr()
    case (LPython::AST::exprType::JoinedStr) : return self().deserialize_JoinedStr();
  File "/home/thirumalai/Open_Source/lpython/src/lpython/python_ast.h", line 10602, in LFortran::LPython::AST::DeserializationBaseVisitor<LFortran::LPython::ASTDeserializationVisitor>::deserialize_JoinedStr()
    v_values.push_back(al, LPython::AST::down_cast<LPython::AST::expr_t>(self().deserialize_expr()));
  File "/home/thirumalai/Open_Source/lpython/src/lpython/python_ast.h", line 10791, in LFortran::LPython::AST::DeserializationBaseVisitor<LFortran::LPython::ASTDeserializationVisitor>::deserialize_expr()
    case (LPython::AST::exprType::FormattedValue) : return self().deserialize_FormattedValue();
  File "/home/thirumalai/Open_Source/lpython/src/lpython/python_ast.h", line 10584, in LFortran::LPython::AST::DeserializationBaseVisitor<LFortran::LPython::ASTDeserializationVisitor>::deserialize_FormattedValue()
    int64_t m_conversion = self().read_int64();
LFortranException: read_int64: Expected integer, got `-`
```
cc @certik

**Comment by certik** [comment]
The last bug is fixed by #515.

**Comment by Thirumalai-Shaktivel** [comment]
The different strings are parsed by the new parser. I believe this Issue can be closed.
